package.json

```diff
@@ -107,10 +107,10 @@
     "ts-node": "10.7.0",
     "tslint": "^6.1.3",
     "tsx": "^3.12.1",
-    "typescript": ">=3.3.1 <5.0.0"
+    "typescript": ">=3.3.1 <4.9.5 || 5.0.1-rc"
```

Comment:
Is there a reason to be so strict here? If we release 4.9.6 (however unlikely), that'll cause problems. That and I see basically no future in which we change the APIs post-RC, so it feels pretty safe to just bump this to `< 5.1.0`.

Though, I also think that you shouldn't specify versions here at all; given ts-eslint prints a very loud warning about using a potentially incompatible TypeScript version, this particular bit IMO only causes pain when trying to get package managers happy in the TypeScript repo or when testing newer releases of TypeScript in other repos...

Reply:
This is the private package.json for our repo, and is not published to the outside world.
Each package that actually depends on TS uses an optional, unbounded peer dependency on TS:
https://github.com/typescript-eslint/typescript-eslint/blob/a2494122fb326cae02d91f3fc1ad37de15039d8e/packages/eslint-plugin/package.json#L78-L82

---

This range here is primarily here as our "top level" range to clearly specify the range for the project. At some point we want to automatically flow this value into our docs and package as the single point-of-definition

Though I agree we should probably keep it as `<5.0.0` instead of dropping it down to `<4.9.5`

Reply:
Ah, yes, oops. I'm conflating this with the fact that prereleases in the tree don't satisfy "*" or "latest" and cause duplication (boo).